system_profile_archiver/delete.py

```diff
@@ -0,0 +1,25 @@
+import psycopg2
+
+from system_profile_archiver import config
+
+DELETE_SQL = "DELETE FROM historical_system_profiles WHERE inventory_id = %s"
+
+
+def delete_by_inventory_id(logger, inventory_id):
+    # TODO: re-use connection and add error handling
+    db_args = {
+        "host": config.DB_HOST,
+        "port": config.DB_PORT,
+        "database": config.DB_NAME,
+        "user": config.DB_USER,
+        "password": config.DB_PASSWORD,
+    }
+    logger.info("opening DB connection")
+    conn = psycopg2.connect(**db_args)
+    cur = conn.cursor()
+    logger.info("deleting records for host %s" % inventory_id)
+    cur.execute(DELETE_SQL, (inventory_id,))
+    conn.commit()
```

Comment:
If the listener errors on execute or commit, does it recover and continue gracefully or bomb out entirely?

Reply:
If something bad happens, an exception will bubble up to the event loop and the message will be dropped.

We should save the message somewhere for reprocessing. I created a card for this.